bundle/config/root.go

```diff
@@ -344,6 +349,14 @@ func (r *Root) MergeTargetOverrides(name string) error {
 		}
 	}
 
+	// Merge `transform.tags`. This field must be overwritten if set, not merged.
+	if v, _ := dyn.Get(target, "transform.tags"); v != dyn.InvalidValue {
+		root, err = dyn.Set(root, "transform.tags", v)
```

Comment:
This should merge into the base instead of overwriting it.

One or more of these transforms can be set at the root level.

Reply:
The challenge with merging is that users then can't override the set of tags anymore. Specifically, users may want to remove the `dev` tag that's added by `mode: development`. So the trade off here is whether we should support removing that vs. the convenience of merging.

Reply:
The major alternative could be to add a special syntax to _remove_ an existing tag? Not sure about a nice way to make that work though :/

```
bundle:
  name: my_bundle

transform:
  tags:
    prod: true

targets:
  dev:
    transform:
      tags:
        ~prod: null # remove both key and value for the prod tag. Unfortunately needs a right-hand side in YML.
```

Reply:
All configuration is additive by default. If you can't define a generic value at the top level then you define them in the target block. We should stick with this for these options and not carve out an exception.

Why would someone configure `prod: true` at the top level?

Reply:
The challenge is that we just need a way to remove tags. Beyond the example above, we always set the `dev` tag in `mode: development`. We need something like the syntax above if we make tags merge rather than overwrite.

Reply:
@pietern so let me know what you think of a syntax for removal. Could be `~dev: null`. Or `~dev: ~`.

Reply:
I looked into supporting something like `~dev: null` and I'm not happy with the complexity that brings. As an alternative I'll just support merging and reserve the `~` character.

Reply:
Discussed: we can potentially add a future setting to presets that disables the `dev` tag behavior instead.